Source/PlasticSourceControl/Private/PlasticSourceControlOperations.cpp

```diff
@@ -262,19 +269,26 @@ bool FPlasticMarkForAddWorker::Execute(FPlasticSourceControlCommand& InCommand)
 	{
 		Parameters.Add(TEXT("-R"));	// needed only at the time of workspace creation, to add directories recursively
 	}
-	// Detect special case for a partial checkout (CS:-1 in Gluon mode)!
-	if (-1 != InCommand.ChangesetNumber)
+	if (InCommand.Files.Num() > 0)
```

Comment:
Similarly, not an issue here, but it would be cleaner to put this at the beginning of the function so it doesn't do anything if not needed